pkg/event/target/webhook.go

```diff
@@ -104,9 +134,52 @@ func (target *WebhookTarget) send(eventData event.Event) error {
 	return nil
 }
 
-// Send - interface compatible method does no-op.
+// IsConnRefusedErr - To check for "connection refused" errors.
+func IsConnRefusedErr(err error) bool {
+	if opErr, ok := err.(*net.OpError); ok {
+		if sysErr, ok := opErr.Err.(*os.SyscallError); ok {
+			if errno, ok := sysErr.Err.(syscall.Errno); ok {
+				if errno == syscall.ECONNREFUSED {
+					return true
+				}
+			}
+		}
+	}
+	return false
+}
+
+// Send - reads an event from store and sends it to webhook.
 func (target *WebhookTarget) Send(eventKey string) error {
-	return nil
+
+	urlStr, pErr := xnet.ParseURL(target.args.Endpoint.String())
+	if pErr != nil {
+		return pErr
+	}
+	_, dErr := net.Dial("tcp", urlStr.Host)
+	if dErr != nil {
+		// To treat "connection refused" errors as errNotConnected.
+		if IsConnRefusedErr(dErr) {
+			return errNotConnected
+		}
+		return dErr
+	}
+
+	eventData, eErr := target.store.Get(eventKey)
+	if eErr != nil {
+		// The last event key in a successful batch will be sent in the channel atmost once by the replayEvents()
+		// Such events will not exist and wouldve been already been sent successfully.
```

Comment:
Typo - `Such events will not exist and wouldve been already been sent successfully.` - >`Such events will not exist and would've already been sent successfully.`

Reply:
Changed